x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AggregateMapper.java

```diff
@@ -131,50 +137,62 @@ private List<IntermediateStateDesc> getNonNull(AggDef aggDef) {
         return l;
     }
 
-    static Stream<Tuple<Class<?>, String>> typeAndNames(Class<?> clazz) {
+    private static Stream<Tuple<Class<?>, Tuple<String, String>>> typeAndNames(Class<?> clazz) {
         List<String> types;
+        List<String> extraConfigs = List.of("");
         if (NumericAggregate.class.isAssignableFrom(clazz)) {
             types = NUMERIC;
         } else if (clazz == Count.class) {
             types = List.of(""); // no extra type distinction
+        } else if (clazz == SpatialCentroid.class) {
```

Comment:
Perhaps check for `SpatialAggregateFunction` instead, since all spatial aggregations will need this.